include/xtensor/xoperation.hpp

```diff
@@ -959,6 +959,74 @@ namespace xt
     {
         return detail::make_xfunction<typename detail::cast<R>::functor>(std::forward<E>(e));
     }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class T>
+    inline auto isin(E&& e, std::initializer_list<T> values) noexcept
+    {
+        auto lambda = [values](const auto& t) -> decltype(e == e) {
```

Comment:
- Is this `decltype` required by some compilers?
- I guess that capturing `values` by value in the lambda is also mandatory?

Reply:
The decltype was my attempt to, in vain, solve some of the compiler issues you address. I'll remove it

Reply:
BTW with again another header I'm more and more longing to a single include